### ISSUE TITLE
UniqueInstrumentMeterCore silently drops observers with the same name and type

### ISSUE THREAD

**Issue by tustvold** [open-telemetry/opentelemetry-rust#541]
The implementation of `UniqueInstrumentMeterCore::new_async_instrument` and `UniqueInstrumentMeterCore::new_sync_instrument` returns the existing instrument if one exists.

This causes issues for value observers where the instruments are not interchangeable as they have different callbacks. I'm not sure what the correct behaviour here is, the [go implementation](https://github.com/open-telemetry/opentelemetry-go/blob/d616df61f5d163589228c5ff3be4aa5415f5a884/metric/registry/registry.go) looks to do something similar, but I can't help feeling silently dropping metrics is perhaps not ideal?

**Comment by jtescher** [comment]
Yeah this is a rougher area still, was discussed in https://github.com/open-telemetry/opentelemetry-specification/issues/1046 with less clear resolution as I recall. Could be worth considering alternatives here.

**Comment by tustvold** [comment]
Thank you for that context, very helpful :+1:

From my perspective, the issue at the moment is that there is no way to detect a collision, and even if you could the returned ValueObserver provides no ability to do anything about it. Perhaps if the API only took a descriptor and then returned a ValueObserver that you could then register one or more callbacks with? I don't know, I'm not sure I completely understand the motivation for making async instrument unique in the first place...

Basically, if there was some way I could get rid of this hack, it would make me very happy :smile: - https://github.com/influxdata/influxdb_iox/pull/1457

**Comment by jtescher** [comment]
This now logs in the new metrics, released in #1156
